### PR TITLE
Closes #2377 Added unit tests

### LINKED ISSUE #2377: General Issue: Add unit test for `dt_level()`
The function `dt_level()`/`dtm_level()` which ared used in the underpinnings of `derive_vars_dt()`/`derive_vars_dtm()`, respectively, needs unit tests

Definition of done: 
Add appropriate unit tests in the `test-derive_dt_dtm_utils` file

### PR DESCRIPTION
**Pull request by StefanThoma**
Added assertion to dt_level function to deal with unexpected inputs

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF version of it to the same folder.
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by StefanThoma** [comment]
I get a strange error in the test suite apparently independent of my code: 

Error in `derive_param_tte()`:
! For some values of "PARAMCD" there is more than one value of "AEDECOD"
ℹ Call `get_one_to_many_dataset()` to get all one-to-many values.

Where the test expects the error as: 
    regexp = paste0(
      "For some values of PARAMCD there is more than one value of AEDECOD.\n",
      "Call `get_one_to_many_dataset()` to get all one to many values."
    )
    
 I can replace this with a snapshot test, maybe!

**Comment by StefanThoma** [comment]
There appears also a deprecation warning: 

Warning ('test-derive_param_tte.R:45:3'): derive_param_tte Test 1: new observations with analysis date are derived correctly ──
`enumerate()` was deprecated in admiraldev 1.1.0.

**Comment by bundfussr** [comment]
> There appears also a deprecation warning:
> 
> Warning ('test-derive_param_tte.R:45:3'): derive_param_tte Test 1: new observations with analysis date are derived correctly ── `enumerate()` was deprecated in admiraldev 1.1.0.

I'm currently trying to fix these in https://github.com/pharmaverse/admiral/issues/2413.

File: R/derive_vars_dt_dtm_utils.R [hunk 1]
```diff
@@ -40,10 +40,14 @@ dtm_level <- function(level) {
 #' @family utils_impute
 #' @keywords internal
 dt_level <- function(level) {
+  possible_values <- c("n", "D", "M", "Y")
+  admiraldev::assert_character_scalar(level, values = possible_values)
```
Comment (bms63) [inline R/derive_vars_dt_dtm_utils.R:44]:
```suggestion
  assert_character_scalar(level, values = possible_values)
```

Comment (bms63) [reply R/derive_vars_dt_dtm_utils.R:44]: we dont use the `package::` convention in admiral

File: tests/testthat/test-dt_level.R
Comment (bms63) [inline tests/testthat/test-dt_level.R:1]: Could you format these like how we do in our other tests - there is an addin in admiraldev that can help you do this automatically.

Comment (StefanThoma) [reply tests/testthat/test-dt_level.R:1]: Thats really cool, I was not aware of this!!

**Review by bms63 — APPROVED** [review]
This looks good - I think the unit test could of kept the expect_error and we coudl use the class argument for it...but this does the same thing with the snapshots i believe...
image

**Pull request merged**